tools/congestion-model/src/model/mod.rs

```diff
@@ -38,7 +38,7 @@ pub struct Model {
 }
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct ShardId(usize);
+pub struct ShardId(pub usize);
```

Comment:
Is this still needed, or just an artefact from previous code iterations? If it's not necessary, I'd prefer this to stay private.

Making it `pub` is okay for our little model if it makes things easier. But generally speaking, it breaks the intended type safety. While the value is private, we can guarantee that `ShardId` can only be created inside this file. So it's easy to check invariants on it. That's about halve the motivation for wrapping the `usize` in a new type. It's the same pattern I used for `QueueId`, `TransactionId`, and `ReceiptId`.

If you just need read-only access to the number, adding a simple getter method, or maybe `impl From<ShardId> for usize {}` or even `impl Deref for ShardId { type Target = usize }` could do the trick without breaking type properties but still allows reading the number with `id.into()` or `*id`, which seems just as convenient as `id.0`.